[16.0][ADD] project_task_link

AungKoKoLin1997 opened the pull request:
This module adds a smart button to the project form view, linking to the tasks associated with the project.

@qrtl

AungKoKoLin1997 committed:
[ADD] project_task_link

yostashiro reviewed (approved):
LGTM. 👍 

![image](https://github.com/OCA/project/assets/7766939/51991159-d72e-4f82-83ca-c5d92512028f)

rinaldifirdaus reviewed (approved):
Functional Test: Works as Expected.

desdelinux reviewed (approved):
Functional review is ok,

but.... 


@AungKoKoLin1997 Can you make a pr odoo with this?

https://github.com/odoo/odoo/pulls

sswapnesh commented:
Hi @AungKoKoLin1997 

![Screenshot (62)](https://github.com/OCA/project/assets/16624719/ec3a869c-f5df-4cc5-8069-80525bd0fb89)


It should use "Name of the tasks" on smart button.

![Screenshot (63)](https://github.com/OCA/project/assets/16624719/92e0c916-24f9-47c1-9679-38fdd56b50ef)

sswapnesh commented on project_task_link/views/project_project_views.xml:
```diff
@@ -0,0 +1,25 @@
+                        <span class="o_stat_value">
+                            <field name="task_count" />
+                        </span>
+                        <span class="o_stat_text">Tasks</span>
```
```suggestion
                        <span class="o_stat_text">
                            <field name="label_tasks" readonly="1"/>
                        </span>
```

AungKoKoLin1997 replied:
Yes. That makes sense to me. Thanks!

AungKoKoLin1997 commented:
> Can you make a pr odoo with this?

@desdelinux I also created PR in odoo repo. https://github.com/odoo/odoo/pull/130976

leemannd commented:
Hello @AungKoKoLin1997 , can you check why the tests are failing?

AungKoKoLin1997 commented:
> can you check why the tests are failing?

@leemannd Tests are failing from `project_template` module. I think we don't need follow up this in current PR.

AungKoKoLin1997 committed:
[FIX] project_task_link
Using Odoo's label_task field within the oe_button_box can lead to conflicts
when other modules reference it. Using a custom field as an alternative ensures
that the original label_task remains untouched, maintaining consistent module compatibility.

leemannd commented:
/ocabo merge nobump

leemannd commented:
/ocabot merge nobump
